### PR TITLE
More Cursors

### PR DESCRIPTION
**Pull request by Maan2003**
This adds pointer and busy cursors. I have only tested this on linux(gtk).

### CODE REVIEW

**Comment by cmyr** [comment]
The cursors that are currently available represent the set of cursors that are implemented on all of our major platforms. I'm not sure how best to expose cursors that are only available on *some* platforms. I do agree that this would be nice, but I would like to be principled about it.

One option might be to have separate `PlatformCursor` enums that describe cursors that are platform specific, and then maybe have a `Cursor::Platform` variant in the main `Cursor` enum? `PlatformCursor` would only be exported if the appropriate `target_os`/`feature`, but this is sort of fragile because of gtk/x11.

what we really want is some way to do like `if `cfg(target_backend = "gtk")`, but that doesn't exist. I'm not sure what to suggest, otherwise...

**Comment by Maan2003** [comment]
So, adding `Cursor::Pointer` should be Ok. Also, I think `Cursor::OpenHand` is not available on windows. 

`Cursor::Busy` is a fundamental cursor IMO :shrug: . How do browsers handle it on Mac OS?

**Comment by cmyr** [comment]
I think macOS only shows the spinner when the main thread is blocked or something?

**Comment by cmyr** [comment]
As far as I can tell there is no pointing hand cursor on windows: ([see docs here?](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-loadcursora))

**Comment by Maan2003** [comment]
> As far as I can tell there is no pointing hand cursor on windows: ([see docs here?](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-loadcursora))

I think `IDC_HAND` is the pointing hand cursor, 
see https://docs.microsoft.com/en-us/windows/win32/menurc/about-cursors#the-mouse-and-the-cursor
> For example, when the cursor crosses over a hyperlink, the system changes the cursor from an arrow to a hand.
> image

**Comment by cmyr** [comment]
Hmm okay, so there's no open hand, for dragging?

**Comment by Maan2003** [comment]
yeah, no open hand :face_with_head_bandage:

**Comment by Maan2003** [comment]
Ready for review. I have removed the busy cursor. Also, I am not sure about CI failure.

**Comment by totsteps** [comment]
Tested on macOS and Windows 10. Works like charm.

File: druid-shell/src/platform/windows/window.rs [hunk 1]
```diff
@@ -1679,7 +1679,9 @@ impl Cursor {
         let name = match self {
             Cursor::Arrow => IDC_ARROW,
             Cursor::IBeam => IDC_IBEAM,
+            Cursor::Pointer => IDC_HAND,
             Cursor::Crosshair => IDC_CROSS,
+            // not correct, maybe polyfill this
```
Comment (cmyr) [inline druid-shell/src/platform/windows/window.rs:1684]: yea, can we open an issue here? Not sure what the best solution is, I'd probably prefer to deprecate the open hand from this enum. 😢

**Review by cmyr — APPROVED** [review]
Looks good!

**Pull request merged**